grpc/federation/cel/url.go

```diff
@@ -0,0 +1,621 @@
+package cel
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/common/types/traits"
+)
+
+const URLPackageName = "url"
+
+var (
+	URLType      = cel.ObjectType("grpc.federation.url.URL")
+	UserinfoType = cel.ObjectType("grpc.federation.url.Userinfo")
+)
+
+func (x *URL) GoURL() (url.URL, error) {
+	var user *url.Userinfo
+	if u := x.GetUser(); u != nil {
+		if u2, err := u.GoUserinfo(); err != nil {
+			return url.URL{}, err
+		} else {
+			user = u2
+		}
+	}
+
+	return url.URL{
+		Scheme:      x.GetScheme(),
+		Opaque:      x.GetOpaque(),
+		User:        user,
+		Host:        x.GetHost(),
+		Path:        x.GetPath(),
+		RawPath:     x.GetRawPath(),
+		OmitHost:    x.GetOmitHost(),
+		ForceQuery:  x.GetForceQuery(),
+		RawQuery:    x.GetRawQuery(),
+		Fragment:    x.GetFragment(),
+		RawFragment: x.GetRawFragment(),
+	}, nil
+}
+
+func (x *Userinfo) GoUserinfo() (*url.Userinfo, error) {
```

Comment:
I believe we don't need to return an error since there is no case to return it? 🤔